zerver/actions/create_user.py

```diff
@@ -189,37 +193,49 @@ def add_new_user_history(user_profile: UserProfile, streams: Iterable[Stream]) -
     recent_message_ids = set(
         Message.objects.filter(
             # Uses index: zerver_message_realm_recipient_id
-            realm_id=user_profile.realm_id,
+            realm_id=realm.id,
             recipient_id__in=recipient_ids,
             date_sent__gt=cutoff_date,
         )
         .order_by("-id")
         .values_list("id", flat=True)[0:MAX_NUM_RECENT_MESSAGES]
     )
 
-    if len(recent_message_ids) > 0:
+    new_onboarding_message_ids = set()
```

Comment:
`new_` is rarely a good way to refer to something that's new to the codebase, because every bit of code eventually becomes old. `tracked_onboarding_message_ids` might be a good name.

Reply:
Updated with `tracked_` prefix.